server/channels/app/mention_results.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+const (
+	// Different types of mentions ordered by their priority from lowest to highest
```

Comment:
How did we decide the priority order for this? It's a bit subjective, I would have had `KeywordMention` as the highest priority for example. Does the order actually matter here?

Reply:
I would've also expected that too since was my intention when it was added, but I think whoever added group mentions might have missed this comment. I don't know if the type of mention actually matters though since I believe the tests are the only thing that care about it. I could see it being useful for certain features like the improved Recent Mentions view we were discussing adding recently, but I guess we can tweak the priority then if we want to actually use it